imod/mf6/ghb.py

```diff
@@ -207,19 +220,54 @@ def from_imod5_data(
         -------
         A  Modflow 6 GeneralHeadBoundary packages.
         """
+        target_top = target_discretization.dataset["top"]
+        target_bottom = target_discretization.dataset["bottom"]
+        target_idomain = target_discretization.dataset["idomain"]
 
         idomain = target_discretization.dataset["idomain"]
         data = {
             "head": imod5_data[key]["head"],
             "conductance": imod5_data[key]["conductance"],
         }
+        is_planar = is_planar_grid(data["conductance"])
 
         if regridder_types is None:
             regridder_types = GeneralHeadBoundaryRegridMethod()
 
         regridded_package_data = _regrid_package_data(
             data, idomain, regridder_types, regrid_cache, {}
         )
+        if is_planar:
+            conductance = regridded_package_data["conductance"]
+
+            planar_head = regridded_package_data["head"]
+            k = target_npf.dataset["k"]
+
+            ghb_alocation = allocate_ghb_cells(
+                allocation_option,
+                target_idomain == 1,
+                target_top,
+                target_bottom,
+                planar_head,
+            )
+
+            layered_head = planar_head.where(ghb_alocation)
+            layered_head = enforce_dim_order(layered_head)
+
+            regridded_package_data["head"] = layered_head
+
+            if "layer" in conductance.coords:
+                conductance = conductance.isel({"layer": 0})
+                conductance = conductance.drop_vars("layer")
```

Comment:
Tip, you can shorten this to:

```suggestion
                conductance = conductance.isel({"layer": 0}, drop=True)
```